pkg/apis/complianceoperator/v1alpha1/complianceremediation_types.go

```diff
@@ -0,0 +1,92 @@
+package v1alpha1
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type RemediationApplicationState string
+
+const (
+	RemediationNotSelected RemediationApplicationState = "NotSelected"
+	RemediationApplied     RemediationApplicationState = "Applied"
+)
+
+type RemediationType string
+
+const (
+	// The remediation wraps a MachineConfig payload
+	McRemediation RemediationType = "MachineConfig"
+)
+
+const (
+	// SuiteLabel defines the label that associates the Remediation with the suite
+	SuiteLabel = "complianceoperator.openshift.io/suite"
+	// ScanLabel defines the label that associates the Remediation with the scan
+	ScanLabel = "complianceoperator.openshift.io/scan"
+)
+
+type ComplianceRemediationSpecMeta struct {
+	// Remediation type specifies the artifact the remediation is based on. For now, only MachineConfig is supported
+	Type RemediationType `json:"type,omitempty"`
+	// Whether the remediation should be picked up and applied by the operator
+	Apply bool `json:"apply"`
+}
+
+// ComplianceRemediationSpec defines the desired state of ComplianceRemediation
+// +k8s:openapi-gen=true
+type ComplianceRemediationSpec struct {
+	ComplianceRemediationSpecMeta `json:",inline"`
+	// The actual remediation payload
+	MachineConfigContents string `json:"machineConfigContents,omitempty"`
```

Comment:
You could make this json.RawMessage and in a subsequent PR we could move this to the MachineConfig type if possible.

Reply:
already moved to MachineConfig